rholang/src/main/scala/rholang/rosette/Equivalence.scala

```diff
@@ -0,0 +1,279 @@
+// -*- mode: Scala;-*- 
+// Filename:    Equivalence.scala 
+// Authors:     Eitan Chatav                                                  
+// Creation:    Thurs Dec 7 11:43:00 2017 
+// Copyright:   See site license 
+// Description: Spatial type checker; see Namespace Logic paper by LGM.
+// ------------------------------------------------------------------------
+
+package coop.rchain.rho2rose
+
+import coop.rchain.syntax.rholang.Absyn._
+import scalaz.{Bind => _, Value => _, _}
+import scalaz.std.list._
+import scalaz.std.option._
+import scala.collection.JavaConverters._
+
+object Equivalences{
+
+  def nameEquivalent(env1: DeBruijn, n1: Chan, env2: DeBruijn, n2: Chan): Boolean = {
```

Comment:
for readability:

```
  def nameEquivalent(env1: DeBruijn, n1: Chan, env2: DeBruijn, n2: Chan): Boolean = (n1, n2) match {
  case (q1: CQuote, q2: CQuote) =>
    (q1.proc_, q2.proc_) match {
      case (pdrop1: PDrop, _) => nameEquivalent(pdrop1.chan_, n2)
      case (_, pdrop2: PDrop) => nameEquivalent(n1, pdrop2.chan_)
      case _                  => structurallyEquivalent(env1,q1.proc_,env2,q2.proc_)
    }
  case (q1: CVar, q2: CVar) => env1.equivalent(q1.var_, env2, q2.var_)
  case _ => false
}
```

Reply:
I don't really like lining up arrows usually because then when you make edits to cases, like say adding a new case, you have to edit other lines just to line things up, and you don't see in the diff only logic changes.

Reply:
That said, if we want to just use `scalafmt` so there are no style fights, I'm down with that.

Reply:
My comment was not regarding the arrow alignment but dealing with pattern matching within pattern matching that was within pattern matching